processor/src/decorators/mod.rs

```diff
@@ -186,6 +188,60 @@ impl Process {
 
         Ok(())
     }
+
+    /// Given a polynomial ( in NTT form ), over quadratic extension field, of domain size
+    /// power of 2, on stack top, this routine interpolates the polynomial using inverse
+    /// Number Theoretic Transform, producing the polynomial in its coefficient form, of
+    /// provided domain size, which is written on the advice tape.
+    ///
+    /// Input stack state should look like
+    ///
+    /// `[result_poly_domain_size, input_poly_domain_size, input_poly_begin_mem_address, ...]`
+    ///
+    /// - `result_poly_domain_size` must be a power of 2.
+    /// - `input_poly_domain_size` must be a power of 2.
+    /// - `result_poly_domain_size <= input_poly_domain_size`
+    /// - Only starting memory address i.e. first code-word's address is provided on stack, consecutive
+    ///  memory addresses are expected to be holding remaining `input_poly_domain_size - 1` many code-words.
+    /// - Each memory address can hold two code-words i.e. each code-word consists of two elements ∈ Z_q | q = `2^64 - 2^32 + 1`
+    ///
+    /// Final advice tape should look like
+    ///
+    /// `[coeff_0, coeff_1, ..., coeff_{n-1}, ...]` | n = result_poly_domain_size
+    ///
+    /// Program which is requesting this non-deterministic computation should read `coeff0`
+    /// first i.e. `coeff{n-1}` should be seen at the very end.
```

Comment:
A couple of small nits about terminology:

- Technically, we are not reading in a polynomial, but evaluations of a polynomial over some domain (another word for this is "codeword"). So, I'd probably start this of as "Given evaluations of a polynomial over the specified domain, interpolates the evaluations into a polynomial in coefficient form and writes the result into the advice tape. The interpolation is performed using the iNTT algorithm. The evaluations are expected to be in the quadratic extension field, and the resulting coefficients are in the quadratic extension field as well."
- A "codeword" is the whole set evaluations rather than just a single evaluation (e.g., all 64 elements are a "codeword"). So, instead of saying something like "Each memory address can hold two code-words...", we can probably say "Each memory address holds two evaluations of a polynomial at adjacent points."

Reply:
Thanks, this reads much better - incorporated changes in f4e857e1